### PR TITLE
Set cost for tests and run parallel tests serially

### PR DESCRIPTION
**Pull request by pfultz2**
This also boosts the number of threads used to try and catch more multithreaded issues.

### CODE REVIEW

**Comment by daniellowell** [comment]
Well, it definitely caught something.
Question is, will #288 resolve these issues.

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -150,83 +150,25 @@ function(add_sanitize_test TEST_SOURCE)
...
-    test_errors.cpp
-    tensor_vec.cpp
-    tensor_test.cpp
-    solver.cpp
-    sequences.cpp
-    kernel_build_params.cpp
-    include_inliner.cpp
-    )
-
-foreach(TEST ${LONG_TESTS})
```
Comment (atamazov) [inline test/CMakeLists.txt:225]: What is the default value of COST? Shall we set is by default for all tests?

Comment (pfultz2) [reply test/CMakeLists.txt:225]: Its zero.

**Comment by pfultz2** [comment]
> Question is, will #288 resolve these issues.

We should probably merge @288 first and see.

File: test/perfdb.cpp [hunk 2]
```diff
@@ -1279,10 +1279,10 @@ struct PerfDbDriver : test_driver
 #if MIOPEN_BACKEND_HIP
             DBMultiThreadedTestWork::threads_count = 20;
 #else
-            DBMultiThreadedTestWork::threads_count = 64;
+            DBMultiThreadedTestWork::threads_count = 128;
```
Comment (JehandadKhan) [inline test/perfdb.cpp:1282]: Why not change the HIP count as well ?

**Comment by daniellowell** [comment]
I merged #288 into develop and develop into this branch.

**Comment by daniellowell** [comment]
Look like we are still seeing deadlocks:
http://micimaster.amd.com/blue/organizations/jenkins/MLLibs%2FMIOpen/detail/mult-thread-tests/7/pipeline/

**Comment by pfultz2** [comment]
Its not a deadlock, it just times out. This could be because of too many threads and processes are pinging the database. If this is 128 processes with 128 threads accessing the DB then that a total 16384 concurrent connections querying the DB at once. Let me lower the number of multiprocess count.

**Pull request merged**